modules/apps/bean-portlet/bean-portlet-cdi-extension/src/main/java/com/liferay/bean/portlet/cdi/extension/internal/BeanPortletExtension.java

```diff
@@ -273,6 +276,35 @@ public void step1BeforeBeanDiscovery(
 		}
 	}
 
+	public <T> void step2ProcessInjectionTarget(
+			@Observes ProcessInjectionTarget<T> processInjectionTarget)
+		throws NoSuchMethodException {
+
+		AnnotatedType<T> annotatedType =
+			processInjectionTarget.getAnnotatedType();
+
+		Class<T> portletClass = annotatedType.getJavaClass();
+
+		if (!Portlet.class.isAssignableFrom(portletClass)) {
+			return;
+		}
+
+		Method destroyMethod = portletClass.getMethod("destroy");
+
+		if (destroyMethod.isAnnotationPresent(PreDestroy.class)) {
```

Comment:
@dantewang Restricting this to portlets that implement the `javax.portlet.Portlet` interface does not take into account a plain pojo bean class that has a method annotated with `@DestroyMethod`